fix: rh-alert not using correct font-family, cursor pointer

brianferry opened the pull request:
Fixes issues related to font-family and cursor pointer for rh-alert described https://github.com/RedHat-UX/red-hat-design-system/issues/516

marionnegp commented on elements/rh-alert/rh-alert.css:
```diff
@@ -73,6 +75,8 @@ header {
 header ::slotted(:is(h1,h2,h3,h4,h5,h6)) {
   margin: 0;
+
+  --rh-font-family-heading: var(--_font-family);
```
The XD kit's version is showing that text in the header should be using Red Hat Text and --rh-font-size-body-text-sm. Font weight's correct.

brianferry replied:
Updated the font size for the header, looks like the font-family was already red hat text?

marionnegp commented on elements/rh-alert/rh-alert.css:
```diff
@@ -52,6 +53,7 @@
   grid-template-columns: min-content 1fr;
   gap: var(--rh-space-xs, 4px);
+  font-family: var(--_font-family);
 }
```
Border width for the container should be --rh-border-width-md for all variants

brianferry replied:
Border width has been updated ✅

marionnegp commented on elements/rh-alert/rh-alert.css:
```diff
@@ -123,15 +127,21 @@ footer ::slotted([slot="actions"]) {
+}
+
 :host(:not([variant])) #container,
 :host([toast]) #container {
```
I believe the token for the box shadow on the toast variant should be --rh-box-shadow-lg. That's the closest to the design specs.

brianferry replied:
Updated this to --rh-box-shadow-lg w/ the token values

marionnegp reviewed (commented):
The changes you made today look good. I found a few things from the previous iteration of rh-alert that weren't consistent with what we have in XD.

brianferry commented:
@marionnegp - Also noticed the margin between the header & the description looked off compared to the UX docs.  In the UX docs it looks like the description is 8px away from the header and 16px away from the action links so I have updated those in the demo as well to use the relevant tokens.

marionnegp commented:
Awesome, @brianferry! The changes look good to me.

Does it matter that the component width changed in the demo?
<img width="960" alt="Screen Shot 2022-09-09 at 1 23 59 PM" src="https://user-images.githubusercontent.com/95588923/189409307-44597b9f-ddf7-4543-accb-a7a096869347.png">

brianferry commented:
@marionnegp - That's a good question, so in the mockups I see it's 396px width and in the component itself we set a max width of 40vh which probably isn't what we want from the demos.  Would we want to set the width to 396px regardless of the content inside so that it always appears the same?  If we set it to a max-width of 396px then we could end up with something like this when we have less content

![image](https://user-images.githubusercontent.com/9708152/189413592-ff48a375-2567-4b25-b555-d72c1b04ee7f.png)

marionnegp commented:
@brianferry, only the toast variant should have a max-width of 555px. The other two variants don't need a max-width.

brianferry commented:
@marionnegp - Got it, I was incorrect in thinking that all the rh-alerts had a max-width of 40vh, that was a demo specific css and not indicative of the component itself so I removed it.  I also added the max-width for the toast variant.

marionnegp commented:
No prob! I wasn't sure at first either, @brianferry.

marionnegp reviewed (commented):
This looks good to me, but I'll let you go ahead and merge it!
